model/task/db.go

```diff
@@ -2633,14 +2633,18 @@ func getTasksByVersionPipeline(versionID string, opts GetTasksByVersionOptions)
 	return pipeline, nil
 }
 
-// FindAllUnmarkedDependenciesToBlock finds tasks that depend on one of the
-// given tasks. For each task dependency, it finds those that have not been
-// marked unattainable and currently have a status that would block the task
-// from running.
+// FindAllDependencyTasksToModify finds tasks that depend on one of the
+// given tasks. The isUnblocking parameter indicates whether we are fetching
+// tasks to unblock them, and if so, for each dependency, all dependency tasks
+// that have been marked unattainable will be retrieved. Otherwise, we are
+// fetching tasks to block them ,and for each task dependency, it finds those
+// that have not been marked unattainable and currently have a status that would
```

Comment:
This last part sounds a little ambiguous because it sounds like the dependent task's status is the one that affects its own blocked state, but the thing that blocks the dependent task is the parent task's status not matching the dependency definition's status filter. Can we clarify that? Maybe providing an example would also help (e.g. a task failed but its dependent task only runs if it succeeds, so the dependent task blocks).

Reply:
I shored this comment up a bit